packages/cinterop/src/darwin/kotlin/io/realm/internal/interop/RealmInterop.kt

```diff
@@ -1127,7 +1127,7 @@ actual object RealmInterop {
                     SyncException(message)
                 }
                 val errorCallback = safeUserData<SyncErrorCallback>(userData)
-                val session = CPointerWrapper(syncSession)
+                val session = CPointerWrapper(realm_clone(syncSession))
```

Comment:
Any reason why this needs to change?

Reply:
Think it is just a miss in the original implementation. @fealebenpae internally commented on https://github.com/realm/realm-core/pull/4951:
```
I understand in pretty much all cases you want to prolong the lifetime of the user object, but I also want to keep the ownership rules clear and consistent: objects returned by calling functions belong to the caller, and objects passed to callbacks are released by default after the callback ends so the implementor of the callback needs to clone them and save them for later.
```
Don't know why it hasn't been crashing before now 🤷

Reply:
Probably the same issue causing #147